turbo/rpchelper/filters.go

```diff
@@ -312,17 +312,27 @@ func (ff *Filters) SubscribeNewHeads(out chan *types.Header) HeadsSubID {
 }
 
 func (ff *Filters) UnsubscribeHeads(id HeadsSubID) bool {
-	ff.mu.Lock()
-	defer ff.mu.Unlock()
-	if ch, ok := ff.headsSubs[id]; ok {
-		close(ch)
-		delete(ff.headsSubs, id)
-		ff.storeMu.Lock()
-		defer ff.storeMu.Unlock()
-		delete(ff.pendingHeadsStores, id)
-		return true
-	}
-	return false
+	ff.mu.RLock()
+	ch, ok := ff.headsSubs[id]
+	ff.mu.RUnlock()
+	if !ok {
+		return false
+	}
+	for {
+		select {
+		case <-ch:
+		default:
+			if ok := ff.mu.TryLock(); ok {
+				close(ch)
```

Comment:
The channel could be closed already (and this may cause panic), between unlocking and re-locking

Reply:
So I suggest that you need to repeat the `ch, ok = ff.headsSubs[id]` under this write lock to make sure you are not trying to close the channel more than once